cmd/fleet/cron.go

```diff
@@ -1064,47 +1064,57 @@ func newAppleMDMDEPProfileAssigner(
 		ctx, name, instanceID, periodicity, ds, ds,
 		schedule.WithLogger(logger),
-		schedule.WithJob("dep_syncer", func(ctx context.Context) error {
-			appCfg, err := ds.AppConfig(ctx)
-			if err != nil {
-				return ctxerr.Wrap(ctx, err, "retrieving app config")
-			}
+		schedule.WithJob("dep_syncer", appleMDMDEPSyncerJob(ds, depStorage, logger)),
+	)
 
-			if !appCfg.MDM.AppleBMEnabledAndConfigured {
-				return nil
-			}
+	return s, nil
+}
 
-			// As part of the DB migration of the single ABM token to the multi-ABM
-			// token world (where the token was migrated from mdm_config_assets to
-			// abm_tokens), we need to complete migration of the existing token as
-			// during the DB migration we didn't have the organization name, apple id
-			// and renewal date.
-			incompleteToken, err := ds.GetABMTokenByOrgName(ctx, "")
-			if err != nil && !fleet.IsNotFound(err) {
-				return ctxerr.Wrap(ctx, err, "retrieving migrated ABM token")
-			}
-			if incompleteToken != nil {
-				if err := apple_mdm.SetABMTokenMetadata(ctx, incompleteToken, depStorage, ds, logger); err != nil {
-					return ctxerr.Wrap(ctx, err, "updating migrated ABM token metadata")
-				}
-				if err := ds.SaveABMToken(ctx, incompleteToken); err != nil {
-					return ctxerr.Wrap(ctx, err, "saving updated migrated ABM token")
-				}
-			}
+func appleMDMDEPSyncerJob(
+	ds fleet.Datastore,
+	depStorage *mysql.NanoDEPStorage,
+	logger kitlog.Logger,
+) func(context.Context) error {
+	var fleetSyncer *apple_mdm.DEPService
+	return func(ctx context.Context) error {
+		appCfg, err := ds.AppConfig(ctx)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "retrieving app config")
+		}
 
-			if fleetSyncer == nil {
-				fleetSyncer = apple_mdm.NewDEPService(ds, depStorage, logger)
+		if !appCfg.MDM.AppleBMEnabledAndConfigured {
+			return nil
+		}
+
+		// As part of the DB migration of the single ABM token to the multi-ABM
+		// token world (where the token was migrated from mdm_config_assets to
+		// abm_tokens), we need to complete migration of the existing token as
+		// during the DB migration we didn't have the organization name, apple id
+		// and renewal date.
+		incompleteToken, err := ds.GetABMTokenByOrgName(ctx, "")
+		if err != nil && !fleet.IsNotFound(err) {
+			return ctxerr.Wrap(ctx, err, "retrieving migrated ABM token")
+		}
+		if incompleteToken != nil {
+			logger.Log("msg", "migrated ABM token found, updating its metadata")
+			if err := apple_mdm.SetABMTokenMetadata(ctx, incompleteToken, depStorage, ds, logger); err != nil {
+				return ctxerr.Wrap(ctx, err, "updating migrated ABM token metadata")
 			}
+			if err := ds.SaveABMToken(ctx, incompleteToken); err != nil {
+				return ctxerr.Wrap(ctx, err, "saving updated migrated ABM token")
+			}
+			logger.Log("msg", "completed migration of existing ABM token")
+		}
 
-			return fleetSyncer.RunAssigner(ctx)
-		}),
-	)
+		if fleetSyncer == nil {
+			fleetSyncer = apple_mdm.NewDEPService(ds, depStorage, logger)
```

Comment:
I'm not sure exactly why this is only created when it reaches this code (instead of creating it outside the job's function), but it was already like this and I was afraid to change it and introduce a subtle issue (cron jobs are not thoroughly tested via the scheduler, oftentimes it's only the inner logic).